test_programs/compile_success_empty/cast_signed_to_u1/src/main.nr

```diff
@@ -0,0 +1,4 @@
+fn main() {
+    let x: i32 = 1;
+    let _ = x as u1;
```

Comment:
Can you expand this a tad?

Might as well add tests asserting e.g. `1 as u1 == 1, 2 as u1 == 0, 3 as u1 == 1`

Reply:
Good idea! I tried it but was lazy to write a test.

Now that I did, it was a bit strange. Casting 2 as i32 to u1 gave me zero, at least when I printed it. But when I compared it with zero it failed. I don't understand why. But now I changed the casting to how it was, except that we now allow again the `i1` type in SSA. That works... but I still don't know why directly casting to `u1` didn't work (I'll take a look later).

Reply:
I'll leave it as it is now. It was working fine before we started validating numeric types, but this is the second time this validation was too restrictive. The previous case was for i128, which doesn't exist in Noir but is used in SSA just fine, and now it's for i1.